rust_src/src/process.rs

```diff
@@ -263,6 +268,32 @@ pub fn process_status(process: LispObject) -> LispObject {
     status
 }
 
+/// Return the connection type of PROCESS.
+/// The value is either the symbol `real', `network', `serial', or `pipe'.
+/// PROCESS may be a process, a buffer, the name of a process or buffer, or
+/// nil, indicating the current buffer's process.
+#[lisp_fn]
+pub fn process_type(process: LispObject) -> LispObject {
+    let mut p;
```

Comment:
Whenever you mark a variable as mutable you should ask why. If the algorithm really requires updates that is expected. In the code below you can use the fact that if/else is also an expression which returns a value.